presto-orc/src/main/java/com/facebook/presto/orc/writer/TimestampColumnWriter.java

```diff
@@ -126,15 +126,28 @@ public void writeBlock(Block block)
                 long value = type.getLong(block, position);
 
                 long seconds = (value / MILLIS_PER_SECOND) - baseTimestampInSeconds;
-                long millis = value % MILLIS_PER_SECOND;
+                long millis = Math.floorMod(value, MILLIS_PER_SECOND);
```

Comment:
One additional comment for the 2nd commit: Add comment here to explain why we have / with floorMod.